### PR TITLE
Move to 18.01

### PR DESCRIPTION
**Pull request by pcm32**
This PR moves both our galaxy container and the setup for using the community images-based setup to Galaxy 18.01, plus our ISA (datatype) and Kubernetes runner changes (better handling of jobs on instance restart). This should be tested with helm charts galaxy version >= 0.4.0 and galaxy-stable version >= 1.5.5 probably.

I would appreciate if people could test this locally and try some tools before we merge.

This PR supersedes #191 to avoid mixing the ISA and Galaxy 18.01 changes and be able to apply them sequentially.

### CODE REVIEW

**Comment by pcm32** [comment]
Further to @djcomlab's testing done on #191, if someone else could test this locally and see their tools working (given that 17.09 to 18.01 is a major version change), I would really appreciate!

Thanks!

**Comment by ilveroluca** [comment]
I'm working on it between talks at a conference.  I've been making some changes to the "simplified" build script to build the containers.  Once it's all up and running I'll test it on minikube.

**Comment by sneumann** [comment]
Hi, I have enabled the Jenkins build for this branch on
http://phenomenal-h2020.eu/jenkins/job/container-galaxy-k8s-runtime/
which should help me testing these containers, as I don't have a local test environment.

**Comment by pcm32** [comment]
These builds have release tags though, which will confuse the release spreadsheet. You would need to change somehow the Jenkins scripts to add a new different prefix. docker build and docker push should allow to test this locally without a Jenkins build.

File: simplified_galaxy_stable_container_creation.sh [hunk 1]
```diff
@@ -14,14 +14,15 @@ then
...
-GALAXY_VER_FOR_POSTGRES=17.09
+
+ANSIBLE_REPO=galaxyproject/ansible-galaxy-extras
+ANSIBLE_RELEASE=18.01-k8s
+GALAXY_VER_FOR_POSTGRES=18.01
 POSTGRES_VERSION=`grep FROM docker-galaxy-stable/compose/galaxy-postgres/Dockerfile | awk -F":" '{ print $2 }'`
+GALAXY_BASE_FROM_TO_REPLACE=quay.io/bgruening/galaxy-base:18.01
 
-TAG=v17.09_cerebellin
+TAG=v18.01-pheno-dev
```
Comment (sneumann) [inline simplified_galaxy_stable_container_creation.sh]: That will become `TAG=v18.01_dalcotidine` when we start the release branch

File: Dockerfile_init [hunk 2]
```diff
@@ -20,6 +20,7 @@ RUN wget -O ephemeris.tar https://api.github.com/repos/galaxyproject/ephemeris/t
 
 #COPY config/galaxy.ini config/galaxy.ini
 COPY config/job_conf.xml config/job_conf.xml
+RUN sed -i 's/HANDLERS -->//' config/job_conf.xml && sed -i 's/<!-- HANDLERS//g' config/job_conf.xml
```
Comment (sneumann) [inline Dockerfile_init]: So these are now correct here ? 
```
pablo Jun 12th at 4:53 PM
Ok, I think I have found the problem... I used to have some sed lines on the Dockerfile to switch handlers for the old setup and the new setup....…
```

Comment (pcm32) [reply Dockerfile_init]: Yes, this lines are correct now... however, image generation will still fail due to https://github.com/bgruening/docker-galaxy-stable/issues/441

We could attempt a fix on our own fork of docker-galaxy-stable.

**Comment by pcm32** [comment]
Ok, containers are building and things work on a fresh minikube setup (0.25). I'm only bothered by https://github.com/bgruening/docker-galaxy-stable/issues/443 but I don't think it should further delay us. Will merge this now.

**Comment by pcm32** [comment]
Apparently we need an additional review given changes, could you do the honours @sneumann? I have tested everything to work.

File: simplified_galaxy_stable_container_creation.sh [hunk 3]
```diff
@@ -1,114 +1,145 @@
...
+fi
 
 
 echo "Relevant containers:"
 echo "Web:          $GALAXY_WEB_K8S_TAG"
 echo "Init Flavour: $GALAXY_INIT_PHENO_FLAVOURED_TAG"
 echo "Postgres:     $POSTGRES_TAG"
 echo "Proftpd:      $PROFTPD_TAG"
+
+if [[ "${DOCKER_PUSH_ENABLED:-}" = "true" ]]; then
```
Comment (pcm32) [inline simplified_galaxy_stable_container_creation.sh]: Isn't this message misleading @ilveroluca ?

Comment (ilveroluca) [reply simplified_galaxy_stable_container_creation.sh]: No, I wouldn't say misleading.  It's downright wrong!  :sweat_smile:.  Sorry.  I'll fix that right away on a new PR

**Pull request merged**